docs(srv): add cron reference to faas caas MTA-4011 (#2535)
scaleway/docs-content: macros/serverless/cron-schedules.mdx

---
macro: cron-schedules
---

> Adding a cron schedule to a specific task allows you to run it automatically at fixed times and dates. Cron jobs are specified using a unix-cron format. You can run a task as often as every minute, up to once a year.

Reviewer: ```suggestion
Adding a cron schedule to a specific task allows you to run it automatically at fixed times and dates. Cron jobs are specified using a unix-cron format. You can run a task as often as every minute, or as infrequently as once a year.
```

## Cron schedule format

Cron jobs are expressed using the `* * * * *` format, which corresponds to the fields below.

```
# ┌───────────── minute
# │ ┌───────────── hour 
# │ │ ┌───────────── day of the month 
# │ │ │ ┌───────────── month 
# │ │ │ │ ┌───────────── day of the week
# │ │ │ │ │ 
# │ │ │ │ │
# │ │ │ │ │
# * * * * *
```

Each field supports a specific range of values.

|       Field      |        Values                        |
|:-----------------|:-------------------------------------|
| Minute           | 0-59                                 |
| Hour             | 0-23                                 |
| Day of the month | 1-31                                 |
| Month            | 1-12 (or JAN-DEC)                    |
| Day of the week  | 0-6 (or SUN-SAT, and 7 for Sunday)   |

<Message type="note">

>   - The month and the day of the wee can be both expressed using numerical or alphabetical Values

Reviewer 2: ```suggestion
  - The month and the day of the week can be both expressed using numerical or alphabetical values
```

| `/`   | Specifies the pace to skip values | `0 */2 * */3 *` runs every 2 hours for a day, once every 3 months.         |

## Cron schedule examples

| Cron Expression Example    | Description                                                                                    |
|----------------------------|--------------------------------------------------------------------------------------------------|
| `0 12 * * *`               | Runs at 12:00 (noon) every day                                                                 |
| `30 9 * * *`               | Runs at 9:30 every day                                                                        |                                                                    |
| `*/5 14-23 * * *`          | Runs every five minutes starting from 14:00 to 23:55 every day                  |
| `0-5 16 * * *`             | Runs every minute from 16:00 to 16:05 every day                          |
| `15,45 12-17 * * WED`      | Runs at minute 15 and 45 from 12:15 to 17:45 every Wednesday                                          |
| `30 9,17 * * MON-FRI`      | Runs at 9:30 and 17:30 from Monday to Friday                          |

> | `1 0 5 * *`                | Runs  the 5th of each month at 00:01                                      |

Reviewer 2: ```suggestion
| `1 0 5 * *`                | Runs the 5th of each month at 00:01                                      |
```

| `0 9 1-24 DEC *`           | Runs at 9:00 every day during the first 24 days of December                             |
| `30 17 * * 5`              | Runs at 17:30 every Friday                                              |

> | `0 0 * 8 *`                | Runs at 00:00 (midnight) every day in august   |

Reviewer 2: ```suggestion
| `0 0 * 8 *`                | Runs at 00:00 (midnight) every day in August   |
```
